src/qililab/qprogram/quantum_machines_compiler.py

```diff
@@ -163,6 +165,7 @@ def compile(
         thresholds: dict[str, float] | None = None,
         threshold_rotations: dict[str, float] | None = None,
         calibration: Calibration | None = None,
+        qdac_buses: list = [],
```

Comment:
1.Don't use mutable types for default argument values. 2. `list` should be annotated to what type it includes. E.g. do `qdac_buses: list[str] | None = None` and initialize `qdac_buses` within the method if needed.

Reply:
And the same comment as in QbloxCompiler applies here too: It's better to validate against a list containing the quantum machines buses.